modules/library/main/src/main/java/org/geotools/filter/visitor/SimplifyingFilterVisitor.java

```diff
@@ -418,7 +421,8 @@ public Object visit(Not filter, Object extraData) {
                 PropertyIsBetween pb = (PropertyIsBetween) simplified;
                 Filter lt = ff.less(pb.getExpression(), pb.getLowerBoundary());
                 Filter gt = ff.greater(pb.getExpression(), pb.getUpperBoundary());
-                return ff.or(lt, gt);
+                Filter nullFilter = ff.isNull(pb.getExpression());
```

Comment:
If the ``featureType`` field has been set, check if the property is nullable before adding this extra. Add a test accordingly.

Reply:
I have just made related changes and added the test as well.